### PR TITLE
kernel-mshv: sync to 5.15.126.mshv9

### PR DESCRIPTION
**Pull request by miz060**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Sync 3.0-dev kernel-mshv to v5.15.126.mshv9

###### Change Log  

- Sync 3.0-dev kernel-mshv to v5.15.126.mshv9

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- BuddyBuild: [547391](https://dev.azure.com/mariner-org/mariner/_build/results?buildId=547391&view=results)

### CODE REVIEW

File: SPECS/kernel-mshv/50_mariner_mshv.cfg [hunk 1]
```diff
@@ -1,16 +1,5 @@
-requirements="Windows/System32/hvix64.exe Windows/System32/hvax64.exe lxhvloader.dll HvLoader.efi"
-for file in $requirements; do
```
Comment (Camelron) [inline SPECS/kernel-mshv/50_mariner_mshv.cfg:2]: Given that this is a azl3 image and therefore generated by grub2-mkconfig, I think this config should stay as it is for 3.0.

File: SPECS/kernel-mshv/kernel-mshv.spec [hunk 2]
```diff
@@ -129,6 +138,20 @@ install -vm 644 vmlinux %{buildroot}%{_libdir}/debug/lib/modules/%{uname_r}/vmli
 # `perf test vmlinux` needs it
 ln -s vmlinux-%{uname_r} %{buildroot}%{_libdir}/debug/lib/modules/%{uname_r}/vmlinux
 
+cat > %{buildroot}/boot/linux-%{uname_r}.cfg << "EOF"
+# GRUB Environment Block
```
Comment (Camelron) [inline SPECS/kernel-mshv/kernel-mshv.spec:142]: This grub environment block is not expected to be used in 3.0. Instead, we use grub2-mkconfig to generate the grub.cfg.

Comment (Camelron) [reply SPECS/kernel-mshv/kernel-mshv.spec:142]: What is the plan for ingesting/creating a MAP AKS-kata image for this? I imagine we will need to take out some of the manual sed'ing steps in AgentBaker and simply rely on the menuentry template in 50_mariner_mshv_menuentry.

**Review by Camelron — CHANGES_REQUESTED** [review]
I think we should sync and discuss the plan/timeline for enabling a 3.0 kata-coco image. 

If we are just trying to get this building right now for this PR, I think we should leave the mkconfig-based stuff (50_mariner_mshv.cfg and 50_mariner_mshv_menuentry) alone as I already tested the functionality when I added it. The AB/MAP changes that come later should be relatively small.

File: SPECS/kernel-mshv/config [hunk 3]
```diff
@@ -1,23 +1,23 @@
 #
 # Automatically generated file; DO NOT EDIT.
-# Linux/x86_64 5.15.126.mshv3 Kernel Configuration
+# Linux/x86_64 5.15.126.mshv9 Kernel Configuration
 #
-CONFIG_CC_VERSION_TEXT="gcc (GCC) 13.2.0"
+CONFIG_CC_VERSION_TEXT="gcc (GCC) 11.2.0"
```
[the remark this answers is not in the document]
Comment (ms-mahuber) [reply SPECS/kernel-mshv/config:5]: Yes, for this PR the config needs no change. We will also align this for kernel-uvm in 3.0-dev

File: SPECS/kernel-mshv/kernel-mshv.spec [hunk 4]
```diff
@@ -20,12 +20,12 @@ Source0:        %{_distro_sources_url}/%{name}-%{version}.tar.gz
 Source1:        config
 Source2:        cbl-mariner-ca-20211013.pem
 Source3:        50_mariner_mshv.cfg
-Source4:        50_mariner_mshv_menuentry
```
Comment (ms-mahuber) [inline SPECS/kernel-mshv/kernel-mshv.spec:23]: we decided to keep this due to grub2

File: SPECS/kernel-mshv/kernel-mshv.spec (hunk 2 above)
Comment (ms-mahuber) [reply SPECS/kernel-mshv/kernel-mshv.spec:142]: discussed with Mitch. This PR was not taking account of the grub2 work that had already happened in 3.0-dev. We'll revisit

File: SPECS/kernel-mshv/50_mariner_mshv.cfg (hunk 1 above)
Comment (miz060) [reply SPECS/kernel-mshv/50_mariner_mshv.cfg]: Good point. Rebased against 3.0-dev branch.

File: SPECS/kernel-mshv/config (hunk 3 above)
[the remark this answers is not in the document]
Comment (miz060) [reply SPECS/kernel-mshv/config]: config change reverted.

File: SPECS/kernel-mshv/kernel-mshv.spec (hunk 4 above)
Comment (miz060) [reply SPECS/kernel-mshv/kernel-mshv.spec:23]: Rebased against 3.0-dev branch.

File: SPECS/kernel-mshv/kernel-mshv.spec (hunk 2 above)
Comment (miz060) [reply SPECS/kernel-mshv/kernel-mshv.spec]: Rebased against 3.0-dev branch.

**Review by Camelron — APPROVED** [review]

**Pull request merged**